sotodlib/core/metadata/obsdb.py

```diff
@@ -81,10 +101,38 @@ def __init__(self, map_file=None, init_db=True):
                     changes = True
             if changes:
                 self.conn.commit()
-
+        self.primary_keys = self.get_primary_keys()
+        if ~np.isin(len(self.primary_keys), [1,3]):
+            raise ValueError("Unsupported number of primary keys in the 'obs' table"
+                             " only 1 or 3 are allowed.")
+
+    def get_primary_keys(self):
```

Comment:
This conflicts with database terminology "primary key", which is the combination of the fields.  So a better name for this function is "get_primary_key_keys" or "get_primary_key_fields".  So how about "get_primary_fields".

Indeed the obsdb code is pretty rigorous in referring to the obsdb columns as "fields", rather than "keys".  So let's keep doing that.